pkg/db/provider.go

```diff
@@ -129,6 +131,15 @@ func (d *DirectoryDB) SearchProviders(criteria types.ProviderSearchParams) ([]*A
 	if criteria.IsMinValidatorPaymentsSet {
 		sb = sb.Where(sb.GE("p.total_paid", criteria.MinValidatorPayments))
 	}
+	if criteria.IsMaxDistanceSet {
+		if !criteria.IsMinRateLimitSet {
+			// we haven't joined on provider id yet
```

Comment:
wonder if the builder is smart enough to ignore repeated identical joins. if not it's probably worth doing a single check on whether any criteria requires meta and join it then. not a blocker

Reply:
yea, I thought about this, and thought about adding a flag, but agree we can do it later when we have more complex logic here.